geoh5py/shared/utils.py

```diff
@@ -624,24 +704,64 @@ def dip_points(points: np.ndarray, dip: float, rotation: float = 0) -> np.ndarra
     return points.T
 
 
-def map_attributes(object_, **kwargs):
+def set_attributes(entity, **kwargs):
+    """
+    Loop over kwargs and set attributes to an entity.
+
+    TODO: Deprecate in favor of explicit attribute setting.
```

Comment:
Maybe you can knock this off if the constructor refactoring I suggested above works out?

Reply:
Can't do until we refactor the Data classes